### DISCUSSION TITLE
Support env-variable FROGMOUTH_CFG_FILE to set the config-location

### DISCUSSION THREAD

**Discussion by woutervh** [Textualize/frogmouth#83] [Q&A]
The config-file is currently hardcoded.
```
def config_file() -> Path:
    """Get the path to the configuration file.

    Returns:
        The path to the configuration file.

    Note:
        As a side-effect, the configuration directory will be created if it
        does not exist.
    """
    (config_dir := xdg_config_home() / ORGANISATION_NAME / PACKAGE_NAME).mkdir(
        parents=True, exist_ok=True
    )
    return config_dir / "configuration.json"
```

This is not virtualenv-friendly.

virtualenv allows to install multiple python-apps in parallel,
but it's problematic when they all want to use the same -configfile in $HOME

pleas support setting this location by env-var:
```
> export FROGMOUTH_CFG_FILE = ....
```

**Comment by davep** [comment]
Frogmouth is written as a standalone application, with no connection to virtual environments and the like. However, it does support the [XDG standard for the location of config and data files](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html); in this case, if you're wanting to configure the location of config and data files for applications run while in a virtual environment, you should find that `XDG_DATA_HOME` and `XDG_CONFIG_HOME` will control the location of the files that Frogmouth looks for and stores.

**Comment by woutervh** [comment]
> Frogmouth is written as a standalone application, with no connection to virtual environments
Every python-package can be installed in a virtualenv.

I don't understand why you close this ticket and move this to Q&A? 

Yes, I'm aware of XDG and no, it is not sufficient, many python-apps have XXX_CFG_FILE env-var or an XXX_HOME env-var with a fallback to XDG-settings.

Are you willing to accept a PR for this?

**Reply by davep** [reply]
I moved it because this was a more appropriate location for the subject, at least as posed. 

Could you explain a bit more about your thinking here, and the motivation? Frogmouth is designed as a stand-alone user application that happens to be written in Python, the implementation language doesn't seem overly important. But I'd love to know what problem you're perceiving here, how a custom location would solve it, and why what's currently provided to achieve that isn't sufficient.

That would be really helpful in understanding things.

**Comment by woutervh** [comment]
I created PR https://github.com/Textualize/frogmouth/pull/84

>Could you explain a bit more about your thinking here,
I install this as a development tool , like black/pytest/pylint ...   to quickly review markdown-files from cli.

all the tools are installed in a virtualenv  alongside the project itself.  
Nothing should write outside the project-directory to minimize intererence with other projects.

**Reply by davep** [reply]
I think this is the thing I'm trying to fully understand: the change you're proposing in your PR only affects the config file, which only contains configuration for light/dark mode, what's a valid Markdown extension, and the last location of the navigation bar; nothing of any real consequence, I would have thought. Meanwhile it won't change how and where the history and the bookmarks are saved (which can also be controlled using the appropriate XDG setting).

If, as you say, *nothing* should write outside the project directory doesn't it make more sense to set `XDG_CONFIG_HOME` and `XDG_DATA_HOME` so that any tool being used in the project, and which follows this standard, will be made to store config and data where you wish? In setting those values not only will frogmouth store where you wish it to, it should also "minimize intererence with other projects" for other tools too.

To be clear: I'm not opposed to what you're after here, I'm just unclear as to why setting the XDG environment variables doesn't achieve your aim and do so in a way that better fits your wish.

**Comment by woutervh** [comment]
>  doesn't it make more sense to set XDG_CONFIG_HOME and XDG_DATA_HOME so that any tool being used in the project

I tried this for a while, but reverted, since it not only affects the project-related tools (good) and all other tools you happen to use when these env-vars are active , esp when running on a linux desktop, your whole desktop tends to get messed up.

I just did not notice the data-dir and the history-file.

**Comment by woutervh** [comment]
I updated the PR
